src/renderer/src/routes/modals/PersistContact/index.tsx

```diff
@@ -0,0 +1,194 @@
+import { useMemo, useState } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import { useTranslation } from 'react-i18next'
+import { MdDeleteForever } from 'react-icons/md'
+import { TbPlus, TbTrash } from 'react-icons/tb'
+import { IContactState, TContactAddress } from '@renderer/@types/store'
+import { BlockchainIcon } from '@renderer/components/BlockchainIcon'
+import { Button } from '@renderer/components/Button'
+import { IconButton } from '@renderer/components/IconButton'
+import { Input } from '@renderer/components/Input'
+import { Separator } from '@renderer/components/Separator'
+import { StyleHelper } from '@renderer/helpers/StyleHelper'
+import { useModalLocation, useModalNavigate } from '@renderer/hooks/useModalRouter'
+import { useAppDispatch } from '@renderer/hooks/useRedux'
+import { ModalLayout } from '@renderer/layouts/Modal'
+import { Contact } from '@renderer/model/Contact'
+import { contactReducerActions } from '@renderer/store/reducers/ContactReducer'
+import { cloneDeep } from 'lodash'
+
+type TFormData = {
+  name: string
+  addresses: TContactAddress[]
+}
+
+type TLocationState = {
+  contact?: IContactState | null
+}
+
+export const PersistContactModal = () => {
+  const { t } = useTranslation('modals', { keyPrefix: 'persistContactModal' })
+  const { t: commonT } = useTranslation('common', { keyPrefix: 'general' })
+
+  const { modalNavigate, modalNavigateWrapper } = useModalNavigate()
+  const location = useModalLocation<TLocationState>()
+  const dispatch = useAppDispatch()
+
+  const contact = useMemo(() => location.state?.contact || null, [location.state?.contact])
+  const [addresses, setAddresses] = useState<TContactAddress[]>(cloneDeep(contact?.addresses) || [])
+  const [showAddressError, setShowAddressError] = useState(false)
+
+  const form = useForm<TFormData>({
+    defaultValues: {
+      name: contact?.name || '',
+    },
+  })
+
+  const contactName = form.watch('name')
+
+  const hasSomeError = Object.keys(form.formState.errors).length > 0
+
+  const handleAddAddress = (address: TContactAddress, addressToEdit?: TContactAddress) => {
+    if (addressToEdit) {
+      addressToEdit.address = address.address
+      addressToEdit.blockchain = address.blockchain
+    } else {
+      setAddresses(prev => [...prev, address])
+    }
+  }
+
+  const handleDeleteAddress = (addressToDelete: TContactAddress) => {
+    if (!contact) return
+    const filterAddress = addresses.filter(address => address !== addressToDelete)
+
+    setAddresses(filterAddress)
+  }
+
+  const openAddAddressModal = (selectedAddress?: TContactAddress) => {
+    if (!contactName) {
+      form.setError('name', { message: t('invalidName') })
+      return
+    }
```

Comment:
You don't need to check this, let the user add a new address even if he hasn't added a name, name validation is already in handleSubmit